code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,268 @@
+	<br>\
+	It is too much. First Phil, then those damn spiders bite Mary too. Heaven help us. <br>\
+	I'll be evacuating as soon as I've grabbed my supplies. I had to pay for those myself. Stupid budget cuts. <br>\
+	The new world they've found for us isn't that hospitable, but at least there are no deadly spid---<br>\
+	(the writing trails off, as if the writer was interrupted)<br>\
+	"
+
+
+/obj/item/weapon/gun/energy/laser/awaymission_aeg
+	name = "Wireless Energy Gun"
+	desc = "An energy gun that recharges wirelessly during away missions. Does not work on the main station."
+	force = 10
+	origin_tech = null
+	self_recharge = 1
+	var/inawaymission = 1
+
+/obj/item/weapon/gun/energy/laser/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == (MAX_Z + 1))
+		if (inawaymission)
+			..()
+		else
+			to_chat(get(src, /mob), "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
+			inawaymission = 1
+	else
+		if (inawaymission)
+			to_chat(get(src, /mob), "<span class='danger'>Your [src] deactivates, as it is out of range from its power source.</span>")
+			power_supply.charge = 0
+			inawaymission = 0
+			update_icon()
+
+
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin
+	name = "beaker 'Black Terror Venom'"
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin/New()
+	..()
+	reagents.add_reagent("terror_black_toxin", 50)
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_green_toxin
+	name = "beaker 'Green Terror Pheromone'"
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin/New()
+	..()
+	reagents.add_reagent("terror_green_toxin", 50)
+	update_icon()
+
+/obj/item/weapon/gun/energy/decloner/awaymission_spidergun
+	name = "Modified Decloner"
+	desc = "A modified biological decloner, tweaked to destroy terror spiders more effectively."
+	icon_state = "decloner"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	projectile_type = "/obj/item/projectile/energy/declone/declone_spider"
+	can_charge = 0
+
+/obj/item/projectile/energy/declone/declone_spider
+	name = "declone"
+	icon_state = "declone"
+	damage = 20
+	damage_type = CLONE
+
+/obj/machinery/computer/id_upgrader
+	name = "ID Upgrade Machine"
+	icon_state = "guest"
+	icon_screen = "pass"
+	//icon = 'icons/obj/computer.dmi'
+	var/list/access_to_give = list()
+
+/obj/machinery/computer/id_upgrader/attackby(obj/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/card/id))
+		var/obj/item/weapon/card/id/I = O
+		if (!access_to_give.len)
+			to_chat(user, "<span class='notice'>This machine appears to be configured incorrectly.</span>")
+			return
+		var/did_upgrade = 0
+		for (var/this_access in access_to_give)
+			if (!(this_access in I.GetAccess()))
+				// don't have it - add it
+				I.access |= this_access
+				to_chat(user, "<span class='notice'>An access type was added to your ID card.</span>")
```

Comment:
`An access type was added to your ID card.`
`An access type was added to your ID card.`
`An access type was added to your ID card.`
`An access type was added to your ID card.`
`An access type was added to your ID card.`
`An access type was added to your ID card.`
`An access type was added to your ID card.`

this should probably be outside of the for loop with a counter of how many accesses it added